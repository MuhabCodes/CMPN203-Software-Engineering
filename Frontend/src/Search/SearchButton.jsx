import React from 'react';
/**
 * This component renders the search button.
 * @component
 * @function SearchButton
 * @returns {null} -displays the search button.
 */
const SearchButton = () => (
  <div className="search-but">
    <button type="submit" id="search-button-btn" data-testid="search-button-test">
      <img src="https://img.icons8.com/ios/25/000000/search--v1.png" alt="" />
      {' '}
    </button>
  </div>
);
export default SearchButton;
