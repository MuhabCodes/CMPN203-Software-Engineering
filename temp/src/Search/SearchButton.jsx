import React from 'react';

const SearchButton = () => (
  <div className="search-but">
    <button type="submit" id="search-button-btn" data-testid="search-button-test">
      <img src="https://img.icons8.com/ios/25/000000/search--v1.png" alt="" />
      {' '}
    </button>
  </div>
);
export default SearchButton;
