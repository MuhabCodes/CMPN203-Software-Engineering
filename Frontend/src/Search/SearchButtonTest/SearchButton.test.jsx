import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import SearchButton from '../SearchButton';
import '@testing-library/jest-dom/extend-expect';

it('it renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchButton />, div);
});

it('renders the button correctly', () => {
  const { getByTestId } = render(<SearchButton />);
  expect(getByTestId('search-button-test')).toBeTruthy();
});

it('matches a snapshot', () => {
  const tree = renderer.create(<SearchButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
