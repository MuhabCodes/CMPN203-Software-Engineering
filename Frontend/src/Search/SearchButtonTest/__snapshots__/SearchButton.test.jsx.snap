// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div
  className="search-but"
>
  <button
    data-testid="search-button-test"
    id="search-button-btn"
    type="submit"
  >
    <img
      alt=""
      src="https://img.icons8.com/ios/25/000000/search--v1.png"
    />
     
  </button>
</div>
`;
